src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,220 @@
+     * @param to The RwaUrn address
+     * @param wad The amount of Dai
+     */
+    event Push(address indexed to, uint256 wad);
+    /**
+     * @notice A contract parameter was updated.
+     * @param what The changed parameter name. Currently the supported values are: "quitTo".
+     * @param data The new value of the parameter.
+     */
+    event File(bytes32 indexed what, address data);
+    /**
+     * @notice The conduit outstanding gem balance was flushed out to `exitAddress`.
+     * @param quitTo The quitTo address.
+     * @param wad The amount flushed out.
+     */
+    event Quit(address indexed quitTo, uint256 wad);
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RwaInputConduit3/not-authorized");
+        _;
+    }
+
+    /**
+     * @notice Define addresses and gives `msg.sender` admin access.
+     * @param _psm PSM contract address.
+     * @param _to RwaUrn contract address.
+     * @param _quitTo Address to where outstanding GEM balance will go after `quit`
+     */
+    constructor(
+        address _psm,
+        address _to,
+        address _quitTo
+    ) public {
+        DSTokenAbstract _gem = DSTokenAbstract(GemJoinAbstract(PsmAbstract(_psm).gemJoin()).gem());
+        psm = PsmAbstract(_psm);
+        dai = DSTokenAbstract(PsmAbstract(_psm).dai());
+        gem = _gem;
+        to = _to;
+        quitTo = _quitTo;
+
+        // Give unlimited approve to PSM gemjoin
+        _gem.approve(address(PsmAbstract(_psm).gemJoin()), type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    /*//////////////////////////////////
+               Authorization
+    //////////////////////////////////*/
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
```

Comment:
Shouldn't we have a file function for `to` as well?

Reply:
hmm...we don't have file function at all for previous version of input conduit

Reply:
Would it makes sense to be able to set `to` in case we update the Urn? Then we don't need to redeploy InputConduit and thus change the target address for the end user.

Reply:
it will make sense only in case of changing the URN, but probably we will have new conduit if have a new URN?

Reply:
Ok, I am fine with that

Reply:
> Would it makes sense to be able to set to in case we update the Urn?

We should probably stick to `pick` here instead of using the `file()` pattern for compatibility sake.